hashmap/hashmap.mbti

```diff
@@ -16,6 +16,7 @@ impl T {
   from_iter[K : Hash + Eq, V](Iter[(K, V)]) -> Self[K, V]
   get[K : Hash + Eq, V](Self[K, V], K) -> V?
   get_or_default[K : Hash + Eq, V](Self[K, V], K, V) -> V
+  get_or_init[K : Hash + Eq, V](Self[K, V], K, (K) -> V) -> V
   is_empty[K, V](Self[K, V]) -> Bool
```

Comment:
Can you align with the old one using `() -> V`, note the callback used to ben an expensive computation so capture the key should not matter that much.
We could provide `get_or_add` as an alternative